addons/mail/static/src/models/thread/thread.js

```diff
@@ -1385,8 +1388,10 @@ function factory(dependencies) {
             if (this.channel_type === 'chat' && this.correspondent) {
                 return this.custom_channel_name || this.correspondent.nameOrDisplayName;
             }
-            if (this.channel_type === 'group') {
-                return this.name || this.members.map(partner => partner.nameOrDisplayName).join(', ');
+            if (this.channel_type === 'group' && !this.name) {
+                const partnerNames = this._sortMembers(this.members).map(partner => partner.nameOrDisplayName);
+                const guestNames = this.guestMembers.map(guest => guest.name);
```

Comment:
It's strange to have a different sorting way for partners and guests. Ideally code for guest should always be the exact mirror as partners (with keeping in mind that some day both should somehow be merged into one in the future).

So in this case, I suppose we should also sort the members field, instead of calling a method on it?